### ISSUE TITLE
Test failure in test-t-sdiv_qrnnd

### ISSUE THREAD

**Issue by orlitzky** [wbhart/flint2#1140]
I just hit this while testing v2.8.5 on Gentoo, with gcc-12.1:

```
2173/2179 Testing: test-t-sdiv_qrnnd
2173/2179 Test: test-t-sdiv_qrnnd
Command: "/var/tmp/portage/sci-mathematics/flint-2.8.5/work/flint2-2.8.5_build/bin/test-t-sdiv_qrnnd"
Directory: /var/tmp/portage/sci-mathematics/flint-2.8.5/work/flint2-2.8.5_build
"test-t-sdiv_qrnnd" start time: May 25 22:12 EDT
Output:
----------------------------------------------------------
sdiv_qrnnd....
<end of output>
Test time =   0.02 sec
----------------------------------------------------------
Test Failed.
"test-t-sdiv_qrnnd" end time: May 25 22:12 EDT
"test-t-sdiv_qrnnd" time elapsed: 00:00:00
```

When I run that test executable directly, I get...

```
$ /path/to/flint2-2.8.5_build/bin/test-t-sdiv_qrnnd
sdiv_qrnnd....Floating point exception
```

and this winds up in my dmesg:

```
[Wed May 25 22:21:48 2022] traps: test-t-sdiv_qrn[27680] trap divide error ip:55c9dc45519f sp:7ffc40eab230 error:0 in test-t-sdiv_qrnnd[55c9dc455000+1000]
```

I'm not sure if gcc-12 has anything to do with it yet. Something similar was reported in https://github.com/wbhart/flint2/issues/1023. It may also be relevant that I had `-ggdb` in my `CFLAGS` and `CXXFLAGS`, since I normally do not.

**Comment by albinahlback** [comment]
Yes, this was addressed by #1135. The problem is that we have code with undefined behavior, which may seem okay in some cases, but really is not due to the compiler optimizing such things away. You will probably see that the problems goes away when setting `-O0`.

**Comment by orlitzky** [comment]
Thanks! I've disabled that one test in our distro package for now, since v2.9.0 is pending and will presumably contain the fix.

**Comment by SnarkBoojum** [comment]
Debian has 2.9.0 which is supposed to fix this, but:
1. I got a [report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1016296) claiming it's back.
2. I could reproduce it.
3. I have the same dmesg error.
